### PR TITLE
Updated data-tools, and created a buld-action

### PR DESCRIPTION
**Pull request by tdonaworth**
## What changed

* Created action to build `data-tools` container
* Updated `data-tools` tests
* Updated `data-tools` delete function to be more safe
* renamed `data.json5` to `portfolio_data.json5`

## Issue

#499

## How to test

`pipenv run pytest`

### CODE REVIEW

File: data-tools/src/import_static_data/import_data.py [hunk 1]
```diff
@@ -41,45 +52,53 @@
...
+def delete_existing_data(conn: sqlalchemy.engine.Engine.connect, data: Dict):
+    for ops_table in data:
         if ops_table not in ALLOWED_TABLES:
             raise RuntimeError("Table not allowed")
-        # nosemgrep: python.sqlalchemy.security.audit.avoid-sqlalchemy-text.avoid-sqlalchemy-text
-        conn.execute(text(f"TRUNCATE TABLE {ops_table} CASCADE;"))
+        # Only truncate if it actually exists
+        if exists(conn, ops_table):
+            # nosemgrep: python.sqlalchemy.security.audit.avoid-sqlalchemy-text.avoid-sqlalchemy-text
+            conn.execute(text(f"TRUNCATE TABLE {ops_table} CASCADE;"))
```
[the remark this answers is not in the document]
Comment (tdonaworth) [reply data-tools/src/import_static_data/import_data.py:80]: Not an issue for this specific use case. White list table list exists, along with static listing.

File: .github/actions/cf-run-task/action.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    description: Application name.
+    required: true
+  CF_COMMAND:
+    description: Task Command to run.
+    required: true
+    default: 'echo "***** NO COMMAND PROVIDED *****"'
+  CF_TASKNAME:
+    description: Name given to the specific task. Can 'grep' this for better log output.
+    required: false
+    default: 'github-action-taks-001'
```
Comment (jonnalley) [inline .github/actions/cf-run-task/action.yml:27]: @tdonaworth I assume you meant `task`

File: data-tools/data/can_data.json5 [hunk 3]
```diff
@@ -123,7 +123,7 @@
     },
     {
       id: 12,
-      number: "G99XXX5",
+      number: "G99XXX7",
```
Comment (jonnalley) [inline data-tools/data/can_data.json5:126]: @tdonaworth is this an intentional change ?

**Review by xlorepdarkhelm — APPROVED** [review]
LGTM

File: data-tools/data/can_data.json5 (hunk 3 above)
Comment (tdonaworth) [reply data-tools/data/can_data.json5:126]: Only to prove data was being changed, so yes.

File: .github/actions/cf-run-task/action.yml (hunk 2 above)
Comment (tdonaworth) [reply .github/actions/cf-run-task/action.yml:27]: just default text, but good catch.

**Review by jonnalley — APPROVED** [review]
engage!

**Pull request merged**